Makefile-docker

```diff
@@ -84,29 +92,32 @@ update_deps_base: ## update the python and node dependencies
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/pip.txt
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/prod.txt
 
-	npm install $(NPM_ARGS)
-	for dest in $(NODE_LIBS_CSS) ; do cp $(NODE_MODULES)$$dest $(STATIC_CSS) ; done
-	for dest in $(NODE_LIBS_JS) ; do cp $(NODE_MODULES)$$dest $(STATIC_JS) ; done
-	for dest in $(NODE_LIBS_JQUERY_UI) ; do cp $(NODE_MODULES)$$dest $(STATIC_JQUERY_UI) ; done
-
 .PHONY: update_deps
 update_deps: update_deps_base ## update the python and node dependencies for development
 	$(PIP_COMMAND) install --progress-bar=off --no-deps --exists-action=w -r requirements/dev.txt
+	npm install $(NPM_ARGS)
 
 .PHONY: update_deps_prod
 update_deps_prod: update_deps_base ## update the python and node dependencies for production
-	npm prune --omit=dev
+	npm install $(NPM_ARGS) --omit=dev --no-save --no-package-lock
```

Comment:
`--no-package-lock` seems extremely dangerous ? How else are we ensuring the hashes are checked ? Also, I assume node of the JS libs we care about in production have dependencies themselves but if we ever start to introduce some, that `--no-package-lock` would make our builds not reproducible...

Reply:
You are right, I refactored this PR already before this comment to make the changes more isolated and discrete. I've added a commit that replaces install for `npm ci` which achieves what I was trying to but in a better way.

I wanted to ensure we rely only on the package-lock.json and that we don't save any files as both reduce reproducibility but you're right I misunderstood what --no-package-lock does and it does the opposite.

This should be much faster/better/secure.